src/panels/MissionsPanel/hooks.ts

```diff
@@ -39,3 +42,31 @@ export function useJumpToTime(fadeTime: number = 1): ReturnType {
     }
   };
 }
+
+export function useSelectedMission(): {
+  mission: Phase | undefined;
+  hasMission: boolean;
+} {
+  const { isInitialized, data, selectedMissionIdentifier } = useAppSelector(
+    (state) => state.missions
+  );
+
+  let mission: Phase | undefined = undefined;
+
+  // If a new mission is added, the missions order might have changed so we try to find
+  // the last one viewed
+  if (isInitialized && selectedMissionIdentifier) {
+    const selectedMission = data.missions.find(
+      (mission) => mission.identifier === selectedMissionIdentifier
+    );
+    if (selectedMission) {
+      mission = selectedMission;
+    } else {
+      mission = data.missions.at(0);
+    }
```

Comment:
```suggestion
    mission = selectedMission ?? data.missions.at(0);
```
?

Reply:
Can we end up here if there are no missions?  Getting and index zero seems a bit dangerous if there's no check against if `data.missions` have enough values